.github/workflows/gcp-deploy-frontend.yaml

```diff
@@ -0,0 +1,216 @@
+        continue-on-error: true
+        run: |
+          gcloud -q run services delete ${{needs.env.outputs.FRONTEND_SERVICE}} \
+            --region ${{needs.env.outputs.CLOUDRUN_REGION}}
+
+  delete-frontend:
+    name: Delete Frontend Container Image
+    runs-on: ubuntu-latest
+    needs: [env]
+
+    steps:
+      - name: Set Up gcloud CLI
+        uses: google-github-actions/setup-gcloud@v0.5.0
+        with:
+          project_id: ${{needs.env.outputs.PROJECT_ID}}
+          service_account_key: ${{secrets.GCP_SA_KEY}}
+          export_default_credentials: true
+
+      - name: "Delete Container Image: ${{needs.env.outputs.FRONTEND_TAG}}"
+        continue-on-error: true
+        run: gcloud -q artifacts docker images delete ${{needs.env.outputs.FRONTEND_TAG}}
+
+  build-frontend:
+    name: Build Frontend Container Image
+    runs-on: ubuntu-latest
+    needs: [env, delete-frontend, undeploy-frontend]
+
+    steps:
+      - name: "Checkout: ${{github.ref_name}}"
+        uses: actions/checkout@v2
+        with:
+          ref: ${{github.event.after}}
+
+      - name: "Build Container Image: ${{needs.env.outputs.FRONTEND_TAG}}"
+        uses: mattes/cached-docker-build-action@v1
+        with:
+          args: |
+            . -f docker/frontend/Dockerfile \
+              --tag ${{needs.env.outputs.FRONTEND_TAG}} \
+              --build-arg GRAPHQL_URL=${{needs.env.outputs.GRAPHQL_URL}}
+          cache_key: "${{github.event.inputs.cache_key}}-${{hashFiles('packages/web/**', 'packages/design-system/**', 'packages/utils/**', 'package.json')}}"
+
+      - name: "Login to Registry: ${{needs.env.outputs.DOCKER_REGISTRY}}"
+        uses: docker/login-action@v1
+        with:
+          registry: ${{needs.env.outputs.DOCKER_REGISTRY}}
+          username: _json_key
+          password: ${{secrets.GCP_SA_KEY}}
+
+      - name: "Push Container Image: ${{needs.env.outputs.FRONTEND_TAG}}"
+        run: docker push ${{needs.env.outputs.FRONTEND_TAG}}
+
+  deploy-frontend:
+    name: Deploy Frontend
+    runs-on: ubuntu-latest
+    needs: [env, build-frontend]
+
+    steps:
+      - name: Set Up gcloud CLI
+        uses: google-github-actions/setup-gcloud@v0.5.0
+        with:
+          project_id: ${{needs.env.outputs.PROJECT_ID}}
+          service_account_key: ${{secrets.GCP_SA_KEY}}
+          export_default_credentials: true
+
+      - name: "Deploy Container Image: ${{needs.env.outputs.FRONTEND_SERVICE}}"
+        run: |
+          gcloud -q run deploy ${{needs.env.outputs.FRONTEND_SERVICE}} \
+            --image ${{needs.env.outputs.FRONTEND_TAG}} \
+            --region ${{needs.env.outputs.CLOUDRUN_REGION}} \
+            --port ${{needs.env.outputs.FRONTEND_PORT}} \
+            --cpu 1 \
+            --memory 512Mi \
+            --ingress all \
+            --max-instances 1 \
+            --allow-unauthenticated \
+            --set-env-vars WEB3_STORAGE_TOKEN=${{secrets.WEB3_STORAGE_TOKEN}} \
```

Comment:
Do we need to be exposing the YouTube API key to the frontend? *(We're not currently for these deployments though maybe it's set in Vercel.)*

Reply:
Not related to this PR, but I actually asked @arsenijesavic to just add it to our repo. It's not really a secret as it is used by the browser. The HTTP referer config for the API key is what secures it.